### PR TITLE
Release v2.4.0

### PR DESCRIPTION
**Pull request by peterk87**
Change log for v2.4.0 release:

- added scheme metadata table to Python package data (#109)
- typing scheme name change from `tb_speciation` to `tb_lineage` (#110)
- updated README (#111)
- updated default low coverage depth value for SE scheme from 50 to 20, max kmer frequency from 1000 to 10,000 and updated help/usage docs (#114)
- improved reverse complement function performance (#118)
- fixed merging of subtyping results with subtype result metadata table (#120)
- added CI testing for Python 3.7, cleaned up setup.py (#119)

### CODE REVIEW

**Comment by DarianHole** [comment]
Hi @peterk87,

Everything seems to be in order on the code side. Pytests run on both 3.6 and 3.7 as expected.

Although I think I've found an installation issue I missed last time (or I don't know how to install). Currently I pip install from the branch (`pip install .`) and when I do that, the aho_corasick and qc directories are not being copied with the data (no issue when I tried 2.3 or 2.2 in this).

I'll see if its on my end a bit more as well

File: bio_hansel/main.py [hunk 1]
```diff
@@ -52,12 +46,12 @@ def init_parser():
...
     parser.add_argument('-s', '--scheme',
                         default='heidelberg',
-                        help='Scheme to use for subtyping (built-in: "heidelberg", "enteritidis", "typhi", "typhimurium", "tb_speciation"; OR user-specified: '
+                        help='Scheme to use for subtyping (built-in: "heidelberg", "enteritidis", "typhi", "typhimurium", "tb_lineage"; OR user-specified: '
                              '/path/to/user/scheme)')
     parser.add_argument('--scheme-name',
                         help='Custom user-specified SNP substyping scheme name')
     parser.add_argument('-M', '--scheme-metadata',
-                        help='Scheme subtype metadata table (CSV or tab-delimited format; must contain "subtype" column)')
+                        help='Scheme subtype metadata table (.TSV format accepted; contain column called "subtype")')
```
[the remark this answers is not in the document]
Comment (peterk87) [reply bio_hansel/main.py]: Made it more explicit in f653c2f6aef0c8a93b2553de9bf926940894bafe with **MUST** and a bit more explanation about accepted file formats

File: README.rst [hunk 2]
```diff
@@ -255,7 +264,18 @@ Analysis of all FASTA/FASTQ files in a directory
...
+Metadata addition to analysis
+-----------------------------
+
+*Works with any of the analyses above
+
+.. code-block:: bash
+
+    hansel -s heidelberg -M <metadata.tsv> -vv --threads <n_cpu> -o results.tab -O match_results.tab -D /path/to/fastas_or_fastqs/
+
+``biohansel`` works best on TSV metadata files. If possible, use a tab separated metadata file or your analysis may fail.
```
[the remark this answers is not in the document]
Comment (peterk87) [reply README.rst]: I've added more documentation here in f653c2f6aef0c8a93b2553de9bf926940894bafe with tables of accepted file formats and a very small example table.

File: setup.py [hunk 3]
```diff
@@ -3,50 +3,47 @@
...
+    description='Subtype microbial whole-genome sequencing (WGS) data using SNV targeting k-mer subtyping schemes.',
+    entry_points={'console_scripts': ['hansel=bio_hansel.main:main']},
+    install_requires=requirements,
+    keywords='Salmonella enterica Heidelberg Enteritidis SNP kmer subtyping Aho-Corasick',
+    license='Apache Software License 2.0',
+    long_description=readme,
+    name='bio_hansel',
+    package_data={'bio_hansel': ['data/*/*.fasta', 'data/*/*.tsv',]},
+    package_dir={'bio_hansel': 'bio_hansel'},
+    packages=find_packages(include=['bio_hansel']),
```
[the remark this answers is not in the document]
Comment (peterk87) [reply setup.py]: Thanks for catching this! Fixed in f653c2f6aef0c8a93b2553de9bf926940894bafe

**Comment by peterk87** [comment]
Thanks a lot @DarianHole for taking a look at this PR! I fixed the packaging issue and made the suggested changes. 

I think it *should* auto-deploy when a release tag is created. 

https://github.com/phac-nml/biohansel/blob/f653c2f6aef0c8a93b2553de9bf926940894bafe/.travis.yml#L13-L23

The user and secure password could be changed to the `phac-nml` and a secure password generated by the travis-ci cli tool. I've added the `phac-nml` user as a project owner on PyPI.

**Comment by DarianHole** [comment]
@peterk87 I am not authorized to merge this one but its good. Let me know if anything else is needed!

**Comment by peterk87** [comment]
Thanks a lot @DarianHole! I'll create a v2.4.0 release and make sure it's up on PyPI

**Comment by glabbe** [comment]
Thanks a lot @peterk87 and @DarianHole for this new version release, much appreciated!

**Pull request merged**